velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -356,4 +356,52 @@ struct ArrayHasDuplicatesFunction {
   }
 };
 
+template <typename TExecParams, typename T>
+struct ArrayNormalizeFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExecParams);
+
+  FOLLY_ALWAYS_INLINE bool callNullFree(
+      out_type<velox::Array<T>>& result,
+      const null_free_arg_type<velox::Array<T>>& inputArray,
+      const null_free_arg_type<T> p) {
+    VELOX_USER_CHECK_GE(
+        p, 0, "array_normalize only supports non-negative p: {}", p);
```

Comment:
Does Presto java give this same error message ?

Reply:
Yes.